walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,55 @@
 
 package com.google.sps;
 
-import java.util.Collection;
+import java.util.Collection; 
+import java.util.Set;
+import java.util.HashSet;
+import java.util.*;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    ArrayList<TimeRange> availableTimes = new ArrayList<TimeRange>();
+    if (request.getDuration() > TimeRange.WHOLE_DAY.duration()) { 
+        // not possible for duration to exceed whole day
```

Comment:
Nit: some indentations in this file look like 4 spaces vs. 2 spaces (they should always be 2 spaces).